### PR TITLE
[gdcvault] Add support for audio extraction

### PR DESCRIPTION
**Pull request by remitamine**
related to this issue https://github.com/rg3/youtube-dl/issues/5784.
about keeping the flv extension for audio it's because when downloading it,ffprobe said that the container is flv but the audio codec is mp3 did i have to change the ext even if it is different from the container.
this is the output of ffprobe for the audio file:
```
Input #0, flv, from 'How to Create a Good Game - From My Experience of Designing Pac-Man-1014631.flv.part':
  Metadata:
    author          : 
    copyright       : 
    description     : 
    keywords        : 
    rating          : 
    title           : 
    presetname      : Custom
    audiodevice     : USB Audio CODEC 
    audiochannels   : 1
    audioinputvolume: 75
    canSeekToEnd    : false
    createdby       : FMS 3.5
    creationdate    : Wed Mar 02 16:30:34 2011
  Duration: 01:12:56.81, start: 0.000000, bitrate: 1 kb/s
    Stream #0:0: Audio: mp3, 22050 Hz, mono, s16p, 32 kb/s
```

### CODE REVIEW

File: youtube_dl/extractor/gdcvault.py [hunk 1]
```diff
@@ -133,16 +146,18 @@ def _real_extract(self, url):
...
+            video_url = 'http://www.gdcvault.com' + direct_url
+            request = compat_urllib_request.Request(video_url, method='HEAD')
+            head = compat_urllib_request.urlopen(request)
 
             return {
                 'id': video_id,
                 'display_id': display_id,
                 'url': video_url,
-                'ext': 'flv',
+                'ext': determine_ext(head.geturl()),
```
Comment (jaimeMF) [inline youtube_dl/extractor/gdcvault.py]: Why is it changed, was it wrong for some url?

File: youtube_dl/extractor/gdcvault.py [hunk 2]
```diff
@@ -133,16 +146,18 @@ def _real_extract(self, url):
             r's1\.addVariable\("file",\s*encodeURIComponent\("(/[^"]+)"\)\);',
             start_page, 'url', default=None)
         if direct_url:
-            video_url = 'http://www.gdcvault.com/' + direct_url
             title = self._html_search_regex(
                 r'<td><strong>Session Name</strong></td>\s*<td>(.*?)</td>',
                 start_page, 'title')
+            video_url = 'http://www.gdcvault.com' + direct_url
+            request = compat_urllib_request.Request(video_url, method='HEAD')
```
Comment (jaimeMF) [inline youtube_dl/extractor/gdcvault.py]: Use [HeadRequest](https://github.com/rg3/youtube-dl/blob/80fb6d4aa47154a1e963b28a17a065dc40a436b8/youtube_dl/utils.py#L1271).

File: youtube_dl/extractor/gdcvault.py [hunk 3]
```diff
@@ -133,16 +146,18 @@ def _real_extract(self, url):
...
+            head = compat_urllib_request.urlopen(request)
```
Comment (jaimeMF) [inline youtube_dl/extractor/gdcvault.py]: Use [_request_webpage](https://github.com/rg3/youtube-dl/blob/80fb6d4aa47154a1e963b28a17a065dc40a436b8/youtube_dl/extractor/common.py#L311)

File: youtube_dl/extractor/gdcvault.py (hunk 1 above)
Comment (remitamine) [inline youtube_dl/extractor/gdcvault.py]: to make it work for both audios and videos.
for example this is for a video page:
```
youtube-dl -F http://gdcvault.com/play/1016925/Bejeweled-Blitz-one-year-in
[GDCVault] Bejeweled-Blitz-one-year-in: Downloading webpage
[info] Available formats for 1016925:
format code  extension  resolution note
0            flv        unknown
```
and this for an audio only page:
```
youtube-dl -F http://gdcvault.com/play/1019442/5-Key-Best-Practices-in
[GDCVault] 5-Key-Best-Practices-in: Downloading webpage
[info] Available formats for 1019442:
format code  extension  resolution note
0            mp3        unknown
```

Comment (jaimeMF) [inline youtube_dl/extractor/gdcvault.py]: Ok

**Comment by remitamine** [comment]
i made the requested changes is there anything else?

**Comment by jaimeMF** [comment]
Cherry picked in 44cae2fb2e7b8cffcdfdd57b1224115345009cc5, thanks.

**Pull request closed without merging**